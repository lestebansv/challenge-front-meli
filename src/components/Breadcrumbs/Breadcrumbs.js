import { useLocation } from 'react-router-dom';
import styles from './Breadcrumbs.module.scss';

/**
 * Component builds the breadcrumb of the product
 * 
 * @param categories (Array) - Product search categories
 * @returns Render view breadcrumbs 
 */
const Breadcrumbs = (categories) => {

    const { search } = useLocation();

    let query = new URLSearchParams(search).get('search');

    if (localStorage.getItem('breadcrumbs')) {
        categories = JSON.parse(localStorage.getItem('breadcrumbs'));
        query = localStorage.getItem('search');
    }

    localStorage.setItem('search', query);
    localStorage.setItem('breadcrumbs', JSON.stringify(categories));

    return <div className={styles.bread}>
        {
            categories.data !== null ? categories.data.map((value) => {
                return `${value} > `
            }) : ''
        }
        <strong>{query}</strong>
    </div>
}

export default Breadcrumbs;