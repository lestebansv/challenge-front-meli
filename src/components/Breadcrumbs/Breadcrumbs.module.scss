@import '../../assets/css/styles.scss';

.bread {
    color: $background-gray;
    font-size: 14px;
    margin: 16px 15%;
}

.bread:last-child {
    font-weight: 600;
}