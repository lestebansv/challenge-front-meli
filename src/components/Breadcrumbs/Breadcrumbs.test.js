import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Breadcrumbs from './Breadcrumbs'

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: () => {
        return {
            pathname: '/items',
            state: {},
            key: '',
            search: 'kindle',
            hash: ''
        }
    }
}));

describe('Breadcrumbs', () => {

    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn()
    };
    
    global.localStorage = localStorageMock;

    it('renders component "Breadcrumbs" with category = Computación', () => {
        const categories = ['Computación'];
        const {container} = render(<Breadcrumbs data={categories}/>);
        expect(container).toMatchSnapshot();
    });

});