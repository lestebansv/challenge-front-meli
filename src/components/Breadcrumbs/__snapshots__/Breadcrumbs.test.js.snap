// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders component "Breadcrumbs" with category = Computación 1`] = `
<div>
  <div
    class="bread"
  >
    Computación &gt; 
    <strong />
  </div>
</div>
`;
